src/notebooks/debugger/helper.ts

```diff
@@ -25,10 +25,18 @@ import ipykernel
 builtins.print(ipykernel.__version__)`;
     const output = await execution.executeHidden(code);
 
-    if (output[0].text) {
-        const version = output[0].text.toString().split('.');
-        const majorVersion = Number(version[0]);
-        return majorVersion >= 6 ? IpykernelCheckResult.Ok : IpykernelCheckResult.Outdated;
+    const versionRegex = /^\d+\.\d+\.\d+$/;
+
+    // It is necessary to traverse all the output to determine the version of ipykernel, some jupyter servers may return extra status metadata
+    for (const line of output) {
+        const lineText = line.text?.toString();
+        if (lineText && versionRegex.test(lineText)) {
+            const [majorVersion] = lineText.split('.').map(Number);
```

Comment:
Why use a regex then ignore the result and do a split instead?

Reply:
You resolved this but it doesn't look like anything changes

Reply:
> You resolved this but it doesn't look like anything changed

Oh hey Rob, I have responsed in channel 3weeks ago and then click the resolve button last week just want to ping you kindly. but somehow my reply disappear. Sorry to confuse you. I will rewrite it.

Reply:
Have changed the regex way to pick up the major version.